### ISSUE TITLE
Catch   Gdk-WARNING 

### ISSUE THREAD

**Issue by apss-pohl** [scorninpc/php-gtk3#51]
Hi,

we get a warning when using https://docs.gtk.org/gtk3/method.Menu.popup_at_pointer.html
```

(php.exe:3064): Gdk-WARNING **: 11:47:21.235: Event with type 4 not holding a GdkSeat. It is most likely synthesized outside Gdk/GTK+

(php.exe:3064): Gdk-WARNING **: 11:47:21.236: Event with type 4 not holding a GdkSeat. It is most likely synthesized outside Gdk/GTK+
```
I think it is not critical but maybe it can be catched?

**Comment by scorninpc** [comment]
And this causing any problem? can you send me a working code to test?

**Comment by apss-pohl** [comment]
Hi Bruno,
we are running more often into issues where such a Critical is thrown but we can't catch those in PHP.
It there a way you can catch them and trow a PHP Error instead?

```php
(php.exe:15616): GLib-GObject-WARNING **: 21:15:33.575: invalid uninstantiatable type '(NULL)' in cast to 'GtkDialog' 
(php.exe:15616): Gtk-CRITICAL **: 21:15:33.575: gtk_dialog_response: assertion 'GTK_IS_DIALOG (dialog)' failed
```
This would be very helpful.

**Comment by scorninpc** [comment]
No you cannot, since this is a internal GTK problem. But its easy to find this problem, as it say `GTK_IS_DIALOG`

You passing some var to some function that need to be a `GtkDialog`. This is not a simple message, this is a problem of your code. Show me a piece of code

**Comment by apss-pohl** [comment]
To find the issue is not the problem, it is already fixed, but to have the ability to writ such into a log file would help because we don't have an open console with our customer. So the errors are invisible to us as the app crashed silently. We need to think of anything else to record them.

**Comment by scorninpc** [comment]
I need to know the code to think about that. But, if the param is a `GtkDialog` and you are passing `GtkEntry`, for example, this is not the extension problem, understand? maybe you can validate the type of variable before call this methods

**Comment by apss-pohl** [comment]
Forget about this issue, it is fixed already :-)
But take it as an example what kind of messages we would like to be able to catch.

**Comment by scorninpc** [comment]
> Forget about this issue, it is fixed already :-)

ok, but i dont think i'm influenced by the title of this issue, or if we are talking about different thinks

The issue is about warning messages, this will not break your code! 

your problem is a critical message, that is logical code problem, not binding problem. 

I understand that we can handle a exception for errors, but in some cases, GTK not will throw a exception, so we need to verify manualy, or in CPP code, or in PHP code. 

Is a nice convension that you may need do some verifications before use. like this https://github.com/scorninpc/php-gtk-cookbook/blob/680771178651ac2998a13c6be8c88c60d64b260f/00024.php#L46 (sorry for exemple, i dont have another one)

**Comment by apss-pohl** [comment]
Yes, sorry, if i hijacked this thread. Topic may be misleading, indeed.
Basically the question is: Can we somehow add functionality to CPP code to "catch  and convert" GTK / GDK output (minimum critical, best any type of message to improve code) to PHP messages and/or writ them to a logfile. This might not be the solution for all issues but it would help a lot.

If this is not possible, then it it what is is :-)

**Comment by scorninpc** [comment]
> Can we somehow add functionality to CPP code to "catch and convert" GTK / GDK output (minimum critical, best any type of message to improve code) to PHP messages and/or writ them to a logfile

i dont know any method to do that. As i know, you need to verify manualy and throw a php exception, like this:

looging parameters count
https://github.com/scorninpc/php-gtk3/blob/b43a78cce8d8d86f4aa745c161a53d8794188b51/src/Gtk/GtkInfoBar.cpp#L24

verify the param type
https://github.com/scorninpc/php-gtk3/blob/65b758e51fb4c3b42d2a187f6012c7d6699ac285/src/Gtk/GtkTreeView.cpp#L42

or verify return
https://github.com/scorninpc/php-gtk3/blob/ebd7d7e7cc02a93565d75557b3c88fa752dd4dc8/src/G/GObject.cpp#L335

is the best scenario, but this will increase considerable the code size, and logic needed, causing more bugs

**Comment by scorninpc** [comment]
i'm working on a new php-gtk version, that will be auto generated! This will works in the right way https://github.com/scorninpc/php8-gtk4-tests

**Comment by scorninpc** [comment]
Sven, this is the thing you may need to know

https://github.com/scorninpc/php-gtk3/pull/57/files#diff-f858f6a50e29bac657d97a796a7a0e7363bbfbae854e1fc50bb927d3aee57330

The `gtk_tree_path_new_from_string` doesn't return boolean when path is empty. This return NULL if the path is invalid

So why the same method on PHP works diferent?

This is a thing you need to verify on PHP, not on PHP, understand?
